src/support.cpp

```diff
@@ -48,6 +48,255 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+            support_infill_part.gradual_infill_areas.clear();
+            support_infill_part.outline = support_islands[i];
+            support_infill_part.insets.clear();
+            // generate insets and use the first inset as the infill area
+            AreaSupport::generateOutlineInsets(
+                support_infill_part.insets,
+                support_infill_part.outline,
+                skin_outline_count,
+                wall_line_width_x);
+            assert(support_infill_part.gradual_infill_areas.size() == 0);
+            if (support_infill_part.insets.empty())
+            {
+                continue;
+            }
+
+            // calculate density areas for this island
+            Polygons less_dense_support = support_infill_part.insets[0]; // one step less dense with each support_step
+            for (unsigned int support_step = 0; support_step < max_support_steps; ++support_step)
+            {
+                size_t min_layer = layer_nr + support_step * gradual_support_step_layer_count + layer_skip_count;
+                size_t max_layer = layer_nr + (support_step + 1) * gradual_support_step_layer_count;
+
+                for (float upper_layer_idx = min_layer; static_cast<unsigned int>(upper_layer_idx) <= max_layer; upper_layer_idx += layer_skip_count)
+                {
+                    if (static_cast<unsigned int>(upper_layer_idx) >= total_layer_count)
+                    {
+                        less_dense_support.clear();
+                        break;
+                    }
+                    const Polygons& relevent_upper_polygons = storage.support.supportLayers[upper_layer_idx].supportAreas;
+                    less_dense_support = less_dense_support.intersection(relevent_upper_polygons);
+                }
+                if (less_dense_support.size() == 0)
+                {
+                    break;
+                }
+
+                // add new gradual_infill_areas for the current density
+                support_infill_part.gradual_infill_areas.emplace_back();
+                std::vector<Polygons>& support_area_current_density = support_infill_part.gradual_infill_areas.back();
+                const Polygons more_dense_support = support_infill_part.insets[0].difference(less_dense_support);
+                support_area_current_density.push_back(more_dense_support);
+            }
+
+            support_infill_part.gradual_infill_areas.emplace_back();
+            std::vector<Polygons>& support_area_current_density = support_infill_part.gradual_infill_areas.back();
+            support_area_current_density.push_back(support_infill_part.insets[0]);
+
+            assert(support_infill_part.gradual_infill_areas.size() != 0 && "support_infill_part.gradual_infill_areas should now be initialized");
+            for (auto tmp_i = 0; tmp_i < support_infill_part.gradual_infill_areas.size(); ++tmp_i)
```

Comment:
'tmp_i' isn't descriptive. Variable names should be descriptive.

Suggested: `part_idx`